src/multimanifestmodule.ts

```diff
@@ -31,7 +32,17 @@ export module multimanifestmodule {
                             options['uri'] = `${Apiendpoint.STACK_API_URL}stack-analyses/?user_key=${Apiendpoint.STACK_API_USER_KEY}`;
                             options['formData'] = payloadData;
                             thatContext = context;
-                            stackanalysismodule.post_stack_analysis(options, file_uri, thatContext, cb);
+
+                            stackAnalysisServices.postStackAnalysisService(options, thatContext)
+                            .then((respData) => {
+                                console.log(`Analyzing your stack, id ${respData}`);
+                                stackanalysismodule.stack_collector_count = 0;
+                                stackanalysismodule.stack_collector(file_uri, respData, cb);
+                            })
+                            .catch((err) => {
+                                console.log(err);
+                                cb(null);
```

Comment:
do we really need to use callback here?

Reply:
I think we can use promise instant of using callback

Reply:
@sudhirverma yes it's needed in catch for the service of the method calling it in case of failure

Reply:
promise would be better to use instead of callback